File: sdks/guides/python/web_scraping/mock_scraper.py
"""Mock scraper - no external API dependencies."""

from datetime import datetime, timezone


def mock_scrape(url: str) -> dict:
    return {
        "url": url,
        "title": f"Page: {url}",
        "content": f"Mock scraped content from {url}. In production, use Firecrawl, Browserbase, or Playwright here.",
        "scraped_at": datetime.now(timezone.utc).isoformat(),
    }


def mock_extract(content: str) -> dict:
    return {
        "summary": content[:80],
        "word_count": str(len(content.split())),
    }

File: examples/typescript/guides/web-scraping/mock-scraper.ts
export interface ScrapeResult {
  url: string;
  title: string;
  content: string;
  scrapedAt: string;
}

export function mockScrape(url: string): ScrapeResult {
  return {
    url,
    title: `Page: ${url}`,
    content: `Mock scraped content from ${url}. In production, use Firecrawl, Browserbase, or Playwright here.`,
    scrapedAt: new Date().toISOString(),
  };
}

export function mockExtract(content: string): Record<string, string> {
  return {
    summary: content.slice(0, 80),
    wordCount: String(content.split(' ').length),
  };
}
